src/models.ts

```diff
@@ -297,3 +304,33 @@ export class AdvancedFilter extends Filter {
     return filter;
   }
 }
+
+export interface IDataReference {
+  target: IFilterTarget;
+}
+
+export interface IEqualsDataReference extends IDataReference {
+  equals: string | boolean | number;
+}
+
+export interface IBetweenDataReference extends IDataReference {
+  between: (string | boolean | number)[];
+}
+
+export interface IValueDataReference extends IDataReference {
+  value: string | boolean | number;
```

Comment:
I was thinking about that too.  I assume the product code which will construct this object will do something like this:

```typescript
const x: IValueDataReference = {
  target: {
    table: "x",
    column: "y"
  },
  value: undefined
};
```

If we make it optional using `value?: ....` then change it to required later it would be a breaking change and I thought it would not hurt to include it now.

Are you worried it would be confusing to have it defined with undefined value?